### PR TITLE
feat: Vault server initialization

### PR DESCRIPTION
**Pull request by soluchok**
<details><summary>Generated OpenAPI spec:</summary>
<p>

```yml
consumes:
- application/json
definitions:
  ErrorResponse:
    description: ErrorResponse to send error message in the response
    properties:
      errMessage:
        type: string
        x-go-name: Message
    type: object
    x-go-package: github.com/trustbloc/edge-service/pkg/restapi/internal/common/http
  authorization:
    properties:
      edv:
        $ref: '#/definitions/location'
      id:
        type: string
        x-go-name: ID
      kms:
        $ref: '#/definitions/location'
      requestingParty:
        type: string
        x-go-name: RequestingParty
      scope:
        properties:
          actions:
            items:
              type: string
            type: array
            x-go-name: Actions
          caveats:
            items:
              properties:
                type:
                  type: string
                  x-go-name: Type
              type: object
            type: array
            x-go-name: Caveats
          target:
            type: string
            x-go-name: Target
          targetAttr:
            type: string
            x-go-name: TargetAttr
        type: object
        x-go-name: Scope
    type: object
    x-go-package: github.com/trustbloc/edge-service/pkg/restapi/vault/operation
  doc:
    properties:
      content:
        type: object
        x-go-name: Content
      edvDocURI:
        type: string
        x-go-name: EDVDocURI
      id:
        type: string
        x-go-name: ID
      tags:
        items:
          type: string
        type: array
        x-go-name: Tags
    type: object
    x-go-package: github.com/trustbloc/edge-service/pkg/restapi/vault/operation
  location:
    properties:
      uri:
        type: string
        x-go-name: URI
      zcap:
        type: object
        x-go-name: ZCap
    type: object
    x-go-package: github.com/trustbloc/edge-service/pkg/restapi/vault/operation
info:
  license:
    name: 'SPDX-License-Identifier: Apache-2.0'
  title: Vault REST API.
  version: 0.1.0
paths:
  /vaults:
    post:
      operationId: createVaultReq
      responses:
        "201":
          $ref: '#/responses/createVaultResp'
        default:
          $ref: '#/responses/genericError'
      summary: Creates a new vault.
      tags:
      - vault
  /vaults/{vaultID}/authorizations:
    post:
      operationId: createAuthorizationReq
      parameters:
      - in: path
        name: vaultID
        required: true
        type: string
        x-go-name: VaultID
      - in: body
        name: Request
        required: true
        schema:
          $ref: '#/definitions/authorization'
      responses:
        "201":
          $ref: '#/responses/createAuthorizationResp'
        default:
          $ref: '#/responses/genericError'
      summary: Creates an authorization.
      tags:
      - vault
  /vaults/{vaultID}/authorizations/{authID}:
    delete:
      operationId: deleteAuthorizationReq
      parameters:
      - in: path
        name: vaultID
        required: true
        type: string
        x-go-name: VaultID
      - in: path
        name: authID
        required: true
        type: string
        x-go-name: AuthorizationID
      responses:
        "200":
          $ref: '#/responses/deleteAuthorizationResp'
        default:
          $ref: '#/responses/genericError'
      summary: Deletes an authorization.
      tags:
      - vault
    get:
      operationId: getAuthorizationReq
      parameters:
      - in: path
        name: vaultID
        required: true
        type: string
        x-go-name: VaultID
      - in: path
        name: authID
        required: true
        type: string
        x-go-name: AuthorizationID
      responses:
        "200":
          $ref: '#/responses/getAuthorizationResp'
        default:
          $ref: '#/responses/genericError'
      summary: Fetches an authorization.
      tags:
      - vault
  /vaults/{vaultID}/docs:
    post:
      operationId: saveDocReq
      parameters:
      - in: path
        name: vaultID
        required: true
        type: string
        x-go-name: VaultID
      - in: body
        name: Request
        required: true
        schema:
          $ref: '#/definitions/doc'
      responses:
        "201":
          $ref: '#/responses/saveDocResp'
        default:
          $ref: '#/responses/genericError'
      summary: Encrypts and stores the document in the vault.
      tags:
      - vault
  /vaults/{vaultID}/docs/{docID}:
    get:
      operationId: getDocReq
      parameters:
      - in: path
        name: vaultID
        required: true
        type: string
        x-go-name: VaultID
      - in: path
        name: docID
        required: true
        type: string
        x-go-name: DocID
      responses:
        "200":
          $ref: '#/responses/getDocResp'
        default:
          $ref: '#/responses/genericError'
      summary: Returns the plaintext document by given ID.
      tags:
      - vault
produces:
- application/json
responses:
  createAuthorizationResp:
    description: createAuthorizationResp model
    schema:
      $ref: '#/definitions/authorization'
  createVaultResp:
    description: createVaultResp model
    schema:
      properties:
        edv:
          $ref: '#/definitions/location'
        id:
          type: string
          x-go-name: ID
        kms:
          $ref: '#/definitions/location'
      type: object
  deleteAuthorizationResp:
    description: deleteAuthorizationResp model
  genericError:
    description: genericError model
    schema:
      $ref: '#/definitions/ErrorResponse'
  getAuthorizationResp:
    description: getAuthorizationResp model
    schema:
      $ref: '#/definitions/authorization'
  getDocResp:
    description: getDocResp model
    schema:
      type: object
  saveDocResp:
    description: saveDocResp model
    schema:
      $ref: '#/definitions/doc'
schemes:
- http
- https
swagger: "2.0"

```

</p>
</details>

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Review by llorllale — APPROVED** [review]
It's missing `DELETE /vaults/{vaultID}`

File: pkg/restapi/vault/operation/operations.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+func New(cfg *Config) (*Operation, error) {
+	return &Operation{}, nil
+}
+
+// GetRESTHandlers get all controller API handler available for this service.
+func (o *Operation) GetRESTHandlers() []support.Handler {
+	return []support.Handler{
+		support.NewHTTPHandler(createVaultPath, http.MethodPost, o.CreateVault),
+		support.NewHTTPHandler(saveDocPath, http.MethodPost, o.SaveDoc),
+		support.NewHTTPHandler(getDocPath, http.MethodGet, o.GetDoc),
```
Comment (llorllale) [inline pkg/restapi/vault/operation/operations.go:41]: This changed to `GET /vaults/{vaultID}/docs/{docID}/metadata` in the latest revision. The reason is we don't have an immediate use case for a user to read back their own documents decrypted.

The document metadata endpoint is a sort of utility method that allows internal orchestration between the components with the aim of hiding some details from the user (eg. EDV doc URIs).

**Comment by soluchok** [comment]
> It's missing `DELETE /vaults/{vaultID}`

Added it here https://github.com/trustbloc/edge-service/pull/573

File: pkg/restapi/vault/operation/operations.go (hunk 1 above)
Comment (soluchok) [reply pkg/restapi/vault/operation/operations.go:41]: Fixed #573

**Pull request merged**